RiotSwiftUI/Modules/Room/Composer/View/Composer.swift

```diff
@@ -193,6 +138,79 @@ struct Composer: View {
             .padding(.bottom, 4)
         }
     }
+
+    private var composerContainer: some View {
+        let rect = RoundedRectangle(cornerRadius: cornerRadius)
+        return VStack(spacing: 12) {
+            if viewModel.viewState.shouldDisplayContext {
+                HStack {
+                    if let imageName = viewModel.viewState.contextImageName {
+                        Image(imageName)
+                            .foregroundColor(theme.colors.tertiaryContent)
+                    }
+                    if let contextDescription = viewModel.viewState.contextDescription {
+                        Text(contextDescription)
+                            .accessibilityIdentifier("contextDescription")
+                            .font(.system(size: 12, weight: .medium))
+                            .foregroundColor(theme.colors.secondaryContent)
+                    }
+                    Spacer()
+                    Button {
+                        viewModel.send(viewAction: .cancel)
+                    } label: {
+                        Image(Asset.Images.inputCloseIcon.name)
+                            .foregroundColor(theme.colors.tertiaryContent)
+                    }
+                    .accessibilityIdentifier("cancelButton")
+                }
+                .padding(.top, 8)
+                .padding(.horizontal, horizontalPadding)
+            }
+            HStack(alignment: .top, spacing: 0) {
+                WysiwygComposerView(
+                    focused: $focused,
+                    content: wysiwygViewModel.content,
+                    replaceText: wysiwygViewModel.replaceText,
+                    select: wysiwygViewModel.select,
+                    didUpdateText: wysiwygViewModel.didUpdateText
+                )
+                .tintColor(theme.colors.accent)
+                .placeholder(viewModel.viewState.placeholder, color: theme.colors.tertiaryContent)
+                .frame(height: wysiwygViewModel.idealHeight)
+                .onAppear {
+                    if wysiwygViewModel.isContentEmpty {
+                        wysiwygViewModel.setup()
+                    }
```

Comment:
This condition is needed to avoid issues with switching from plain to rich that would lose the content of the composer. 
It is definitely not ideal, we probably need a better way to initialise the composer state (perhaps we just need the Rust model to never ever be totally empty of text nodes 🤔)